odo/core.py

```diff
@@ -57,12 +60,15 @@ def _transform(graph, target, source, excluded_edges=None, ooc_types=ooc_types,
                ooc_types=ooc_types)
     try:
         for (A, B, f) in pth:
+            log.debug('[{src} -> {dest}] {f} ({kwargs})'.format(f=f.__name__,
```

Comment:
in general you should let logging libraries handle string formatting, this ensuers that no formatting happens unless the log message is being handled. `logging` uses `%` formatting so you can pass the values to format in as positionals after the message format.